src/librustc/ty/layout.rs

```diff
@@ -1737,10 +1754,12 @@ impl<'a, 'tcx, C> TyLayoutMethods<'tcx, C> for Ty<'tcx>
                     } else {
                         tcx.mk_mut_ref(tcx.lifetimes.re_static, nil)
                     };
-                    return cx.layout_of(ptr_ty).map_same(|mut ptr_layout| {
-                        ptr_layout.ty = this.ty;
-                        ptr_layout
-                    });
+                    return MaybeResult::from(
+                        cx.layout_of(ptr_ty).to_result().map(|mut ptr_layout| {
```

Comment:
Even with this much indentation, rustfmt will keep `return MaybeResult::from(cx.layout_of(ptr_ty).to_result().map(` on one line and just have the closure expression indented by itself, e.g.:
```rust
                    return MaybeResult::from(cx.layout_of(ptr_ty).to_result().map(
                        |mut ptr_layout| {
                            ptr_layout.ty = this.ty;
                            ptr_layout
                        },
                    ));
```

Reply:
@eddyb actually rustfmt does this
```rust
return MaybeResult::from(cx.layout_of(ptr_ty).to_result().map(|mut ptr_layout| {
        ptr_layout.ty = this.ty;
        ptr_layout
    }));
```

Reply:
Only if you have less indentation than you actually do here.